### ISSUE TITLE
Timeout listing datasets within VS Code but CLI operating quickly

### ISSUE THREAD

**Issue by strattondrs** [zowe/vscode-extension-for-zowe#1081]
**Describe the bug**

Using the Zowe CLI to list the datasets matching a certain pattern returns in a handful of seconds (3-4), but querying with the VS Code extension on the same pattern is resulting in a timeout.  This appears to be size related, in that patterns that return smaller sizes do work.

**To Reproduce**

CLI

Issue the following PowerShell command:
`measure-command { $output = zowe files ls ds 'DAN1.*'; $output | Out-Default; $output | Measure-Object | Out-Default }`

VS Code

1. Open Zowe extension icon
2. Click magnifying glass against the default Zowe profile 
3. Enter DAN1.* as pattern
4. Timeout received

**Expected behavior**

The list of datasets populated in VS code to appear and not timeout

**Desktop (please complete the following information):**

- OS: Windows 10 20H2 OS Build 19042.630
- Zowe Explorer Version: v1.10.1
- (Optional) Zowe CLI Version: 6.24.5
- (Optional) Are you using Secure Credential Store? No

**Additional context**

Only 108 datasets match the criteria.  Client resides in Brisbane, Australia (GMT+10), while Mainframe is in the US (GMT-5).  Ping ~220ms time.

Smaller sets work OK, no obvious issues on timeouts with jobs or USS files.

Result from Powershell
```
Count             : 108
Average           :
Sum               :
Maximum           :
Minimum           :
StandardDeviation :
Property          :

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 2
Milliseconds      : 958
Ticks             : 29582606
TotalDays         : 3.42391273148148E-05
TotalHours        : 0.000821739055555556
TotalMinutes      : 0.0493043433333333
TotalSeconds      : 2.9582606
TotalMilliseconds : 2958.2606
```

Error from VS Code
```
Retrieving response from zowe.List Error: z/OSMF REST API Error:
Rest API failure with HTTP(S) status 500
rc:       4
reason:   1
category: 2
message:  receiveResponseHeader: timeout receiving response (>30 secs): TsoServerConnection(USER=DAN1, ASID=0x0052, QID=0x00090005)
```

**Comment by crawr** [comment]
Hi @strattondrs, the squad took a look at how well Zowe Explorer supports listing large datasets. You can view the results of that research here: [issue 327](https://github.com/zowe/vscode-extension-for-zowe/issues/327#issuecomment-578083803) and [https://ent.box.com/s/vk3oqr5porksaio0up4ooekcgiamruek](https://ent.box.com/s/vk3oqr5porksaio0up4ooekcgiamruek) where the expected time to list 30k members should take only up to 5s. 

Perhaps this is network related? Please let us know if you are still encountering this issue repeatedly or if we can close it.

**Comment by strattondrs** [comment]
Yes, I'd seen that issue prior to raising when trying to understand why it wasn't working.  The problem still remains for us, and is reproducible across multiple machines.  

Updating latest CLI (6.26.0) and Zowe Explorer (1.12.1) has no impact on behaviour, we're still seeing fast behaviour from a CLI and timeouts in the VS Code extension.  Running against another list which returns 632 instead of 108 takes ~0.5 seconds longer from the CLI (3.3 seconds vs 2,8 seconds).

Network may be part of it, as ping times to the mainframe through a VPN tunnel sit around 240ms.  It's located in Dallas, we're located in Australia.  However, Windows VMs on a box physically co-located in the data centre with the mainframe also exhibit this behaviour, and from the CLI, takes roughly the same time (perhaps up to 200ms faster on the larger list).

Pulling a 6MB member down via FTP from Oz I get 270KB/second.  The same from the co-located VM is 11.5MB/sec.

**Comment by crawr** [comment]
Hi @strattondrs , in a more recent version of Zowe CLI, there is a new `responseTimeout` parameter that will help you set the maximum amount of time in seconds the z/OSMF Files TSO servlet should run before returning a response. Please let us know if this solves your issue.

**Comment by JillieBeanSim** [comment]
fixed in PR https://github.com/zowe/vscode-extension-for-zowe/pull/1914
